### PR TITLE
Update severity of certain checks in `Plugin_Readme_Check`

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
This PR address first bullet point in #253 issue.

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: tests/phpunit/Checker/Checks/Plugin_Readme_Check_Tests.php [hunk 1]
```diff
@@ -154,17 +159,17 @@ public function test_run_md_with_errors() {
...
-		$this->assertEquals( 'stable_tag_mismatch', $warnings['readme.md'][0][0][3]['code'] );
+		// Check for trunk stable tag error.
+		$this->assertArrayHasKey( 0, $errors['readme.md'] );
+		$this->assertArrayHasKey( 0, $errors['readme.md'][0] );
+		$this->assertArrayHasKey( 'code', $errors['readme.md'][0][0][0] );
+		$this->assertEquals( 'trunk_stable_tag', $errors['readme.md'][0][0][0]['code'] );
+
+		// Check for stable tag mismatch file error.
+		$this->assertArrayHasKey( 0, $errors['readme.md'] );
+		$this->assertArrayHasKey( 0, $errors['readme.md'][0] );
```
Comment (swissspidy) [inline tests/phpunit/Checker/Checks/Plugin_Readme_Check_Tests.php]: The exact same assertions are already done above, so these are redundant.

Comment (mukeshpanchal27) [reply tests/phpunit/Checker/Checks/Plugin_Readme_Check_Tests.php]: Thanks @swissspidy for review. Please check attached screenshot to know how we get the error/notice data.

Screenshot 2023-09-25 at 2 56 31 PM

Comment (swissspidy) [reply tests/phpunit/Checker/Checks/Plugin_Readme_Check_Tests.php]: That's not what I am referring to.

These two lines are literally duplicated:

```php
$this->assertArrayHasKey( 0, $errors['readme.md'] );
$this->assertArrayHasKey( 0, $errors['readme.md'][0] );
```

Lines 162-163 and 169-170

Comment (swissspidy) [reply tests/phpunit/Checker/Checks/Plugin_Readme_Check_Tests.php]:
```suggestion
```

Comment (mukeshpanchal27) [reply tests/phpunit/Checker/Checks/Plugin_Readme_Check_Tests.php]: Ohh gotcha. Updated in https://github.com/10up/plugin-check/pull/279/commits/c723ec824f5e1130e0646f79e1c87ae9789d85ff

**Review by swissspidy — APPROVED** [review]

**Review by felixarntz — APPROVED** [review]
Thank you @mukeshpanchal27, LGTM!

**Pull request merged**